fars/booking/views.py

```diff
@@ -104,31 +105,28 @@ def get(self, request, bookable):
         booking.end = dateutil.parser.parse(request.GET['et']) if 'et' in request.GET else booking.start + timedelta(hours=1)
         booking.bookable = self.context['bookable']
         booking.user = request.user
-        form = BookingForm(instance=booking)
+        form = get_form_class(booking.bookable.metadata_form)(instance=booking)
         self.context['form'] = form
 
-        if _is_admin(request.user, self.context['bookable']):
+        if _is_admin(request.user, booking.bookable):
             self.context['repeatform'] = RepeatingBookingForm()
-        self.context['metadataform'] = self.context['bookable'].get_metadata_form()
 
         return render(request, self.template, context=self.context)
 
 
     def post(self, request, bookable):
-        form = BookingForm(request.POST, instance=Booking())
+        booking = Booking()
+        booking.user = self.context['user']
+        booking.bookable = self.context['bookable']
+        form = get_form_class(booking.bookable.metadata_form)(request.POST, instance=booking)
         self.context['form'] = form
-        metadataform = self.context['bookable'].get_metadata_form(request.POST)
 
         if form.is_valid():
             booking = form.instance
-            if metadataform:
-                self.context['metadataform'] = metadataform
-                if metadataform.is_valid():
-                    booking.metadata = json.dumps(metadataform.cleaned_data)
-                else:
-                    return render(request, self.template, context=self.context, status=400)
+            print(form.cleaned_data)
```

Comment:
Should this `print` statement still be here?

Reply:
Fixed